### PR TITLE
Build test contracts from solidity

### PR DESCRIPTION
**Pull request by emaG3m**
This is something I've been wanting to do for a while. Compiling directly from solidity gives us the flexibility to change and alter smart contracts as we find new test cases. having a copy and pasted, hard to read abi (and obvivously reading bytecode) doesn't give us much leverage or knowledge over the smart contract we're testing. I want to get more logs in our zombie contract so I can test logs more effectively.

Thinking maybe the result of the build `artifacts/zombieAttack.bin & artifacts/zombieAttack.abi` should be gitignored and required on the part of the dev.... can delete the files for the PR wondering what you think. 

**Related Issue(s):**

**Closes Issue(s):** 

**Nearcore Branch Info:**  e9a9fc02c19ed2b97f720242bf98338639ed520c 

**Unit tests completed?:** 

**Relevant README/Documentation Updated?:** N

### CODE REVIEW

File: test/build.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
```
Comment (emaG3m) [inline test/build.sh:1]: this is what we're doing in `near/near-evm` which makes it easier to use/deploy web3 contracts, as opposed to using truffle contracts

File: test/contracts/SafeMath.sol [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+library SafeMath {
+    /**
+     * @dev Returns the addition of two unsigned integers, reverting on
+     * overflow.
+     *
+     * Counterpart to Solidity's `+` operator.
+     *
+     * Requirements:
+     * - Addition cannot overflow.
```
Comment (barbaraliau) [inline test/contracts/SafeMath.sol:11]: You can't tell Addition what to do!

File: test/build.sh (hunk 1 above)
Comment (barbaraliau) [reply test/build.sh:1]: cool beans

File: test/build.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+contracts=(
+    "ZombieAttack"
+)
+
+truffle compile || exit 1
+for contractName in "${contracts[@]}"
```
Comment (barbaraliau) [inline test/build.sh:8]: We are only using these for testing, right? If so, let's move the compiled files from the root level `artifacts/` and into `test/artifacts/`.

Comment (emaG3m) [reply test/build.sh:8]: doe the same go for `near_evm.wasm`?

Comment (barbaraliau) [reply test/build.sh:8]: I think so? Do we only use that for testing? If so, then just move that entire directory

**Comment by emaG3m** [comment]
@barbaraliau ok moved artifacts directory. 

how do you feel about having the dev building their own abi/bin?? (by running `./build.sh` before running their tests) it will likely only need to be done once unless the contracts are changed. I included the build in near-evm, but one day edited the contracts and forgot to build them, pushed, so the contracts got out of whack with their build, so I decided to have just have them built on the part of the dev. pros and cons.

**Comment by barbaraliau** [comment]
How long does it take to build?

That might be a good idea. Then we don't have to worry about keeping that
up to date.

**Comment by emaG3m** [comment]
@barbaraliau takes probably 2.6 seconds to build (lolz it's not like we're building rust here) . They would need to rebuild if the contracts are updated (con) but being responsible for the build ensures that contracts/build never get out of whack by accident in a bad push to master.

**Comment by emaG3m** [comment]
force pushed all changes in single commit due to an awkward rebase

**Pull request merged**